docs/source/tutorial_external_solver.rst

```diff
@@ -0,0 +1,382 @@
+          # Note how this uses the buffer protocol setup to allow efficient
+          # data access as a Numpy Vector
+          result = np.frombuffer(self._impl, dtype=np.float)
+          if ensure_copy:
+              result = result.copy()
+          return result
+
+      def _copy_data(self):
+          # This uses the second exposed 'init' signature to delegate to the C++ copy constructor
+          self._impl = Vector(self._impl)
+
+      def _scal(self, alpha):
+          self._impl.scal(alpha)
+
+      def _axpy(self, alpha, x):
+          self._impl.axpy(alpha, x._impl)
+
+      def dot(self, other):
+          return self._impl.dot(other._impl)
+
+      def l1_norm(self):
+          raise NotImplementedError
+
+      def l2_norm(self):
+          return math.sqrt(self.dot(self))
+
+      def l2_norm2(self):
+          return self.dot(self)
+
+      def sup_norm(self):
+          raise NotImplementedError
+
+      def dofs(self, dof_indices):
+          raise NotImplementedError
+
+      def amax(self):
+          raise NotImplementedError
+
+
+The implementation of the `WrappedVectorSpace` is very short as most of the necessary methods
+of |VectorSpace| are implemented in :class:`~pymor.vectorarrays.list.ListVectorSpace`.
+
+.. jupyter-execute::
+
+  class WrappedVectorSpace(ListVectorSpace):
+
+      def __init__(self, dim):
+          self.dim = dim
+
+      def zero_vector(self):
+          return WrappedVector(Vector(self.dim, 0))
+
+      def make_vector(self, obj):
+          # obj is a `model.Vector` instance
+          return WrappedVector(obj)
+
+      def __eq__(self, other):
+          return type(other) is WrappedVectorSpace and self.dim == other.dim
+
+
+Wrapping the `model.DiffusionOperator` is straightforward as well. We just need to attach
+suitable |VectorSpaces| to the class and implement the application of the operator on a |VectorArray|
+as a sequence of applications on single vectors.
+
+.. jupyter-execute::
+
+  class WrappedDiffusionOperator(Operator):
+      def __init__(self, op):
+          assert isinstance(op, DiffusionOperator)
+          self.op = op
+          self.source = WrappedVectorSpace(op.dim_source)
+          self.range = WrappedVectorSpace(op.dim_range)
+          self.linear = True
+
+      @classmethod
+      def create(cls, n, left, right):
+          return cls(DiffusionOperator(n, left, right))
+
+      def apply(self, U, mu=None):
+          assert U in self.source
+
+          def apply_one_vector(u):
+              v = Vector(self.range.dim, 0)
+              self.op.apply(u._impl, v)
+              return v
+
+          return self.range.make_array([apply_one_vector(u) for u in U._list])
+
+
+Putting it all together
+-----------------------
+
+As a demonstration, we will use our (stationary) toy diffusion solver in a transient example with timestepping
+provided by pyMOR. For details on the discretization setup, see :doc:`tutorial01`. First up, we implement
```

Comment:
```suggestion
provided by pyMOR. First up, we implement
```

I don't think that this tutorial really is a good reference, as much is said that isn't needed here and many other things are missing.